src/alcotest/core.mli

```diff
@@ -48,6 +48,7 @@ module type S = sig
     ?and_exit:bool ->
     ?verbose:bool ->
     ?compact:bool ->
+    ?max_log_lines:int option ->
```

Comment:
`None` could as well be `Unlimited`, in which case there's no collapsing of `int option option` into `int option`, it would be just `limit option` i.e. `?max_log_lines:limit` where `type limit = Unlimited | Limit of int`. It's different for `log_dir` since there is always a log directory but not always a lines limit. Now, unlike for the log dir, the CLI doesn't let us specify all the possible values for the lines limit. For example, if the default was a limit of 100, we could not set it to unlimited via the CLI. Maybe that's the issue that should be fixed. Maybe we should support something like `--max-log-lines=unlimited` (= `--tail-errors=unlimited`). The default would still be unlimited but the user/reader wouldn't have to worry about it.

Reply:
I implemented these things: support for `--tail-errors=unlimited` and clearer type `[`Unlimited | `Limit of int]`.

Reply:
You're right about the comparison to `log_dir`; that was a bad example. I was trying to make the point that the runtime API should not allow an assertion of 'unlimited' if the CLI API does not. 

Personally, I interpret `?max_log_lines:int` as already having a 
 already has mechanism for asserting the lack of a limit in either not passing the option or `?max_log_lines=None` (the use of an optional argument being a syntactic convenience for that assertion). However, I can see your interpretation too.

I suggest either collapsing the option or adding a `--tail-errors=unlimited` feature

Reply:
You're right about the comparison to `log_dir`; that was a bad example. I was trying to make the point that the runtime API should not allow an assertion of 'unlimited' if the CLI API does not.

Personally, I interpret `?max_log_lines:int` as already having a mechanism for asserting the lack of a limit: either not passing the option or passing `?max_log_lines=None` (the use of an optional argument being a syntactic convenience for that assertion). However, I can see your interpretation too.

I suggest either adding a `--tail-errors=unlimited` feature or collapsing the options to unify the `Core` and `Cli` APIs,  whichever you prefer.

Reply:
The solution in https://github.com/mirage/alcotest/pull/200/commits/fbec296e685edda6d9458a2fbace81280a442837 works well; thanks!